EIPS/eip-7867.md

```diff
@@ -0,0 +1,565 @@
+within a single batch MUST be contiguous and applied atomically (or the batch
+rolled back.)
+
+Loose atomicity, on the other hand, is a weaker guarantee. In the event of a
+block reorg, any number of calls from the batch MAY appear on chain (possibly
+interspersed with other transactions). If there are no block reorgs, loose
+atomicity MUST provide the same guarantees as strict.
+
+The none level of atomicity only provides the guarantee that the calls appear on
+chain in the order they are in the batch. Any number of calls from the batch
+MAY appear on chain (possibly interspersed with other transactions).
+
+### Behaviour
+
+#### `wallet_sendCalls`
+
+The wallet MUST reject `wallet_sendCalls` requests with error code
+`MISSING_CAP` where both:
+
+ * the batch-scope `flowControl` capability **is not** present; and
+ * a call-scope `flowControl` capability **is** present.
+
+Note that the above requirement still applies if the call-scope `flowControl`
+capability is marked as optional.
+
+When `flowControl` is present in the batch-scope `capabilities`, the following
+changes override the behaviour specified in EIP-5792.
+
+##### Removed Requirements
+
+These requirements defined in EIP-5792 are removed:
+
+> The wallet:
+>
+> * MUST NOT await for any calls to be finalized to complete the batch
+> * MUST submit multiple calls as an atomic unit in a single transaction
+> * MAY revert all calls if any call fails
+> * MUST not execute any further calls after a failed call
+> * MAY reject the request if one or more calls in the batch is expected to
+>   fail, when simulated sequentially
+
+##### Added Requirements
+
+The wallet:
+
+###### Batch Atomicity
+
+* MAY break the batch over multiple transactions.
+* MUST treat a missing batch-scope `atomicity` level as equivalent to `strict`.
+* MUST provide strict guarantees (as defined above) when the batch-scope
+  `atomicity` is `strict`.
+* MUST provide _at least_ loose guarantees (as defined above) when the
+  batch-scope `atomicity` is `loose`.
+* MUST provide _at least_ the in-order call inclusion guarantee (as defined
+  above) when the batch-scope `atomicity` is `none`.
+* MAY provide loose guarantees (as defined above) when the batch-scope
+  `atomicity` is `none`.
+* MAY provide strict guarantees (as defined above) when the batch-scope
+  `atomicity` is `loose` or `none`.
+* MUST rollback the batch if one or more critical calls (as defined above) fail.
+* MUST NOT rollback the batch if zero critical calls (as defined above) fail.
+  * In other words, if the only failures are non-critical, the successful calls
+    have to appear on chain.
+* MUST NOT execute a call (or ever allow a call to be executed) more than once.
+
+###### Flow Control
+
+* MUST treat a missing call-scope `flowControl` capability as equivalent to
+  setting `onFailure` to `rollback`.
+* MUST treat a missing call-scope `onFailure` mode as equivalent to `rollback`.
```

Comment:
I think this is another argument for request-level `onFailure` settings - as an example I don't think `loose` and `none` are necessarily capable of `rollback`?